web/meetup-facilitator/src/components/Recommendation.jsx

```diff
@@ -0,0 +1,217 @@
+// ... (imports)
+import RecGoogleMap from './RecGoogleMap';
+import RecPopup from './RecPopup'; // Import RecPopup component
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useParams } from "react-router-dom";
+
+const extract_coordinates_from_google_maps_url = (url) => {
+  const match = url.match(/@(-?\d+\.\d+),(-?\d+\.\d+)/);
+
+  if (match && match.length === 3) {
+    const [_, latitude, longitude] = match;
+    return { lat: parseFloat(latitude), lng: parseFloat(longitude) };
+  }
+
+  return null;
+};
+
+const Recommendation = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedDiv, setSelectedDiv] = useState(null);
+  const [recommendations, setRecommendations] = useState([]);
+  const [sampleCoordinates, setSampleCoordinates] = useState([]);
+  const [popupTimes, setPopupTimes] = useState([]);
+  const [votesData, setVotesData] = useState([]);
+  const [selectedTime, setSelectedTime] = useState(null);
+  const {group_id} = useParams();
+  const userdata = {
+    'user_id': localStorage.getItem('email')
+  } 
+  //handle on select recpopup
+  const onSelectHandler = (selectedTime) => {
+    updateVotes(selectedTime, recommendations[selectedDiv].id);
+  };
+
+  //update vote when time is selected
+  const updateVotes = (selectedTime, recommendationId) => {
+    const index = recommendations.findIndex((rec) => rec.id === recommendationId);
+    const updatedRecommendations = [...recommendations];
+    updatedRecommendations[index].votes = (updatedRecommendations[index].votes || 0) + 1;
+    setRecommendations(updatedRecommendations);
+    setSelectedTime(selectedTime);
+  };
+
+  useEffect(() => {
+    // Fetch votes data
+    fetch(import.meta.env.VITE_SERVER + `groups/${group_id}/votes`, {
+      method: "GET",
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        setVotesData(json);
+      })
+      .catch((error) => {
+        console.error('Error fetching votes:', error);
+      });
+  }, []);
+
+  useEffect(() => {
+    // Fetch recommendations data
+    fetch(import.meta.env.VITE_SERVER + `groups/${group_id}/recs`, {
+      method: "GET",
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        setRecommendations(json);
```

Comment:
When parsing the json recommendations, I'm afraid it doesn't parse recursively, so the `times` json array doesn't get parsed to an array and remains as a string. For this reason, we'll have to manually parse all the `times` fields.
```
json.map((rec) => {
  rec.times = JSON.parse(rec.times);
});
```
Something like this before setting the recommendations should do it.